<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>RNA-Seq using Nextflow: Day 2 â€“ nextflow_to_rstudio</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../lesson.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">RNA-Seq using Nextflow: Day 2</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html">Home</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../setup.html">Setup</a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-nfcore-rnaseq-to-rstudio" role="button" data-bs-toggle="dropdown" aria-expanded="false">nfcore-rnaseq to RStudio</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-nfcore-rnaseq-to-rstudio">    
        <li>
    <a class="dropdown-item" href="../notebooks/1-recap_Day1.html">
 <span class="dropdown-text">RECAP Day 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/recap_casestudy.html">
 <span class="dropdown-text">Recall the case study</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/1.2_Learning_objectives.html">
 <span class="dropdown-text">Learning objectives</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/2_nextflow_to_RStudio.html">
 <span class="dropdown-text">Setting up R/RStudio on Pawsey Nimbus</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-identifying-differentially-expressed-(de-genes" role="button" data-bs-toggle="dropdown" aria-expanded="false">Identifying differentially expressed (DE) genes</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-identifying-differentially-expressed-(de-genes">    
        <li>
    <a class="dropdown-item" href="../notebooks/3.1_exploratory_analysis.html">
 <span class="dropdown-text">Exploratory analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/4-DE_analysis.html">
 <span class="dropdown-text">Create the DESeqDataSet object in DeSeq2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/4-DE_analysis_2.html">
 <span class="dropdown-text">Identify differentially expressed (DE) genes</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-enrichment-analysis" role="button" data-bs-toggle="dropdown" aria-expanded="false">Enrichment analysis</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-enrichment-analysis">    
        <li>
    <a class="dropdown-item" href="../notebooks/6_Whatis_Enrichment_analysis.html">
 <span class="dropdown-text">What is enrichment analysis?</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/7_R_DEGenes_to_functional_enrichments_clusProf.html">
 <span class="dropdown-text">DE genes to functional enrichments</span></a>
  </li>  
    </ul>
  </li>
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setting-up-rrstudio-on-pawsey-nimbus" id="toc-setting-up-rrstudio-on-pawsey-nimbus" class="nav-link active" data-scroll-target="#setting-up-rrstudio-on-pawsey-nimbus"><strong>Setting up R/RStudio on Pawsey Nimbus</strong></a>
  <ul class="collapse">
  <li><a href="#questions" id="toc-questions" class="nav-link" data-scroll-target="#questions"><strong>Questions</strong></a></li>
  <li><a href="#objectives" id="toc-objectives" class="nav-link" data-scroll-target="#objectives"><strong>Objectives</strong></a></li>
  <li><a href="#r-and-rstudio" id="toc-r-and-rstudio" class="nav-link" data-scroll-target="#r-and-rstudio"><strong>R and RStudio</strong></a>
  <ul class="collapse">
  <li><a href="#using-rrstudio-ide-for-de-analysis" id="toc-using-rrstudio-ide-for-de-analysis" class="nav-link" data-scroll-target="#using-rrstudio-ide-for-de-analysis"><strong>Using R/RStudio IDE for DE analysis</strong></a></li>
  </ul></li>
  <li><a href="#run-rstudio-on-the-nimbus-trainee-vm" id="toc-run-rstudio-on-the-nimbus-trainee-vm" class="nav-link" data-scroll-target="#run-rstudio-on-the-nimbus-trainee-vm"><strong>Run RStudio on the Nimbus trainee VM</strong></a>
  <ul class="collapse">
  <li><a href="#step1-run-the-rserver-command-pawsey-nimbus" id="toc-step1-run-the-rserver-command-pawsey-nimbus" class="nav-link" data-scroll-target="#step1-run-the-rserver-command-pawsey-nimbus"><strong>Step1: Run the rserver command (Pawsey Nimbus)</strong></a></li>
  <li><a href="#step2-open-rstudio-from-a-browser-local-machine" id="toc-step2-open-rstudio-from-a-browser-local-machine" class="nav-link" data-scroll-target="#step2-open-rstudio-from-a-browser-local-machine"><strong>Step2: Open RStudio from a browser (Local machine)</strong></a></li>
  <li><a href="#transfer-files-to-correct-path" id="toc-transfer-files-to-correct-path" class="nav-link" data-scroll-target="#transfer-files-to-correct-path"><strong>Transfer files to correct path</strong></a></li>
  <li><a href="#working-in-rstudio" id="toc-working-in-rstudio" class="nav-link" data-scroll-target="#working-in-rstudio"><strong>Working in RStudio</strong></a></li>
  <li><a href="#r-markdown" id="toc-r-markdown" class="nav-link" data-scroll-target="#r-markdown"><strong>R-markdown</strong></a></li>
  <li><a href="#the-correct-gene-count-matrix" id="toc-the-correct-gene-count-matrix" class="nav-link" data-scroll-target="#the-correct-gene-count-matrix"><strong>The correct gene-count matrix</strong></a></li>
  <li><a href="#open-the-r-markdown-file" id="toc-open-the-r-markdown-file" class="nav-link" data-scroll-target="#open-the-r-markdown-file"><strong>Open the R-markdown file</strong></a></li>
  <li><a href="#load-the-r-libraries" id="toc-load-the-r-libraries" class="nav-link" data-scroll-target="#load-the-r-libraries"><strong>Load the R-libraries</strong></a></li>
  <li><a href="#set-the-current-working-directory" id="toc-set-the-current-working-directory" class="nav-link" data-scroll-target="#set-the-current-working-directory"><strong>Set the current working directory</strong></a></li>
  <li><a href="#import-the-gene-count-matrix" id="toc-import-the-gene-count-matrix" class="nav-link" data-scroll-target="#import-the-gene-count-matrix"><strong>Import the gene-count matrix</strong></a></li>
  <li><a href="#key-points" id="toc-key-points" class="nav-link" data-scroll-target="#key-points"><strong>Key points</strong></a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">



<section id="setting-up-rrstudio-on-pawsey-nimbus" class="level1">
<h1><strong>Setting up R/RStudio on Pawsey Nimbus</strong></h1>
<section id="questions" class="level3 questions">
<h3 class="anchored" data-anchor-id="questions"><strong>Questions</strong></h3>
<ul>
<li>How to get the full gene count-matrix for downstream analysis?</li>
<li>How to open RStudio on the Pawsey training VM?</li>
<li>How to import a count-matrix in R?</li>
</ul>
</section>
<section id="objectives" class="level3 objectives">
<h3 class="anchored" data-anchor-id="objectives"><strong>Objectives</strong></h3>
<ul>
<li>Introduction to RStudio</li>
<li>Learn to start RStudio on Pawsey Nimbus</li>
<li>Load the count matrix in R</li>
</ul>
</section>
<section id="r-and-rstudio" class="level2">
<h2 class="anchored" data-anchor-id="r-and-rstudio"><strong>R and RStudio</strong></h2>
<ul>
<li>R is a programming language for statistical computing and graphics.</li>
<li>RStudio is an integrated development environment which can be used to write and excecute R-code.</li>
</ul>
<section id="using-rrstudio-ide-for-de-analysis" class="level3">
<h3 class="anchored" data-anchor-id="using-rrstudio-ide-for-de-analysis"><strong>Using R/RStudio IDE for DE analysis</strong></h3>
<ul>
<li>The gene-count matrix which was generated using the nefcore-rnaseq pipline can be used to perform statistical analyses and determine differentially expressed (DE) genes and pathways.</li>
<li>Multiple independant packages/libraries have been developed in <a href="https://www.r-project.org/">R-programming</a>, which can be used for performing various kinds of â€˜omicsâ€™ analysis.</li>
<li>R packages such as <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DeSeq2</a>, <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">EdgeR</a> etc are used for identification of differencially expressed (DE) genes. Today we will be using DeSeq2.</li>
</ul>
</section>
</section>
<section id="run-rstudio-on-the-nimbus-trainee-vm" class="level2">
<h2 class="anchored" data-anchor-id="run-rstudio-on-the-nimbus-trainee-vm"><strong>Run RStudio on the Nimbus trainee VM</strong></h2>
<p>This is a two step process</p>
<section id="step1-run-the-rserver-command-pawsey-nimbus" class="level3">
<h3 class="anchored" data-anchor-id="step1-run-the-rserver-command-pawsey-nimbus"><strong>Step1: Run the rserver command (Pawsey Nimbus)</strong></h3>
<div class="sourceCode" id="cb1"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a># Create a temporary rstudio-server  folder on your instance:</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>mkdir -p /tmp/rstudio-server</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>PASSWORD='abc' singularity exec -B /tmp/rstudio-server:/var/lib/rstudio-server -B /tmp/rstudio-server:/var/run/rstudio-server -B ~/base_directory/working_directory:/home /cvmfs/containers.biocommons.aarnet.edu.au/r/n/rnaseq_rstudio.sif rserver --auth-none=0 --auth-pam-helper-path=pam-helper --server-user ubuntu</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="callout-note callout callout-style-simple callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
<strong>No output</strong> is good output <span class="emoji" data-emoji="grinning">ðŸ˜€</span>
</div>
</div>
<div class="callout-body-container callout-body">
<p>Once you paste the above command in the termonal window and press <code>Enter</code>, you <strong>SHOULD NOT</strong> see any output (the command prompt is stuck at the terminal, awaiting any further action!). If this happens, we are good to go..</p>
</div>
</div>
<div class="callout-note callout callout-style-simple callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
For advanced command-line users
</div>
</div>
<div class="callout-body-container callout-body">
<p>You can also run the above command using a <code>screen</code> window.</p>
</div>
</div>
</section>
<section id="step2-open-rstudio-from-a-browser-local-machine" class="level3">
<h3 class="anchored" data-anchor-id="step2-open-rstudio-from-a-browser-local-machine"><strong>Step2: Open RStudio from a browser (Local machine)</strong></h3>
<ul>
<li><p>Open up a browser window (<strong>IMPORTANT</strong>: Firefox does not work. Use Chrome or Safari.)</p></li>
<li><p>Type <code>146.118.XX.XX:8787</code> in your browser where the XX.XX will be replaced by your IP specific digits. So if my login IP is 146.118.67.219, I will type <code>146.118.67.219 :8787</code> in the browser (e.g.&nbsp;<code>Chrome</code>) and press enter.</p></li>
<li><p>Enter the username, which is your image operating system, which is <code>ubuntu</code>.</p></li>
<li><p>Enter the password, which in this example is <code>abc</code>.</p></li>
<li><p>To end the session, simply exit from the browser.</p></li>
<li><p>To also end the session on your Pawsey Nimbus, run the following command:</p></li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>lsof -ti:8787 | xargs kill -9</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="transfer-files-to-correct-path" class="level3">
<h3 class="anchored" data-anchor-id="transfer-files-to-correct-path"><strong>Transfer files to correct path</strong></h3>
<p>Please transfer the files <code>Full_count_matrix.txt</code> and<code>rnaseq_DE_Full_matrix.Rmd</code> from the path <code>~/base_directory/working_directory/</code> to the path <code>~/base_directory/working_directory/rstudio/</code> by running the following two commands</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>cp ~/base_path/working_directory/Full_count_matrix.txt/ rstudio/</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>cp ~/base_path/working_directory/rnaseq_DE_Full_matrix.Rmd rstudio/</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>We will discuss what these files contain in the sections below.</p>
</section>
<section id="working-in-rstudio" class="level3">
<h3 class="anchored" data-anchor-id="working-in-rstudio"><strong>Working in RStudio</strong></h3>
</section>
<section id="r-markdown" class="level3">
<h3 class="anchored" data-anchor-id="r-markdown"><strong>R-markdown</strong></h3>
<ul>
<li>R Markdown is a file format for making dynamic documents with R.</li>
<li>An R Markdown document is written in markdown (an easy-to-write plain text format) and contains chunks of embedded R code, like the document shown below.</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">RStudio interface</figcaption><p></p>
<p><img src="../fig/R_markdown.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="the-correct-gene-count-matrix" class="level3">
<h3 class="anchored" data-anchor-id="the-correct-gene-count-matrix"><strong>The correct gene-count matrix</strong></h3>
<p>In yesterdayâ€™s session, we had generated a gene-count matrix was generated using the <code>nf-core/rnaseq</code> pipeline. This file can be seen in the path:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>~/base_directory/working_directory/results/star_salmon/salmon.merged.gene_counts.tsv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The input to the <code>nf-core/rnaseq</code> pipeline were sub-setted fastq files from the 6 samples and these were mapped to a specific region of <code>chr18</code> of the mouse genome. This was done so that the excecution of the <code>nf-core/rnaseq</code> pipeline can be demonstrated on our small-sized training VMs <code>(2 cpus and 8 GB RAM)</code> in practical time <code>(~ 20 mins)</code>. <br><strong>However</strong>, for todayâ€™s session, we will be using a pre-computed gene-count matrix which was generated by aligning the full set of reads to the complete mouse genome (version mm10). This gene-count matrix file was downloaded previously along with the other files from cloudstor and is named as <code>Full_count_matrix.txt</code>.</p>
</section>
<section id="open-the-r-markdown-file" class="level3">
<h3 class="anchored" data-anchor-id="open-the-r-markdown-file"><strong>Open the R-markdown file</strong></h3>
<p><img src="../fig/Rmarkdown_open_file.png" class="img-fluid"></p>
<p>The Rmarkdown file <code>rnaseq_DE_Full_matrix.Rmd</code> contains all R-scipts required for todayâ€™s analysis. Each individual chunk of code can be run separately by clicking on the arrow next to it top left corner. <br>*<strong>NOTE</strong> <br> Please use this <code>.rmd</code> file for excecuting the code (<strong>recommended</strong>), but make sure that you run the code chunks in a sequential order. <br>Alternatively you can make your own new markdown (.rmd) file and insert R-code chunks into it <span class="emoji" data-emoji="grimacing">ðŸ˜¬</span></p>
</section>
<section id="load-the-r-libraries" class="level3">
<h3 class="anchored" data-anchor-id="load-the-r-libraries"><strong>Load the R-libraries</strong></h3>
<ul>
<li>Import all the R libraries/packages which will assist us at various steps in the analysis.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>({</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"DESeq2"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"edgeR"</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"limma"</span>)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"RColorBrewer"</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"gplots"</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"factoextra"</span>)</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"devtools"</span>)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"rstudioapi"</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tibble"</span>)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"pheatmap"</span>)</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"biomaRt"</span>)</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"annotables"</span>)</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"org.Mm.eg.db"</span>)</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"biobroom"</span>)</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"clusterProfiler"</span>)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"pathfindR"</span>)</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="set-the-current-working-directory" class="level3">
<h3 class="anchored" data-anchor-id="set-the-current-working-directory"><strong>Set the current working directory</strong></h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>current_path <span class="ot">&lt;-</span> <span class="fu">getActiveDocumentContext</span>()<span class="sc">$</span>path </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="fu">dirname</span>(current_path ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="import-the-gene-count-matrix" class="level3">
<h3 class="anchored" data-anchor-id="import-the-gene-count-matrix"><strong>Import the gene-count matrix</strong></h3>
<p>The DESeq2 package expects count data from RNA-seq or another high-throughput sequencing experiment in the form of a matrix of integer values, as input. The rows in the matrix correspond to genes, and the columns correspond to samples. The values in the matrix should be un-normalized counts.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the Full count matrix</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>counttable_original<span class="ot">&lt;-</span><span class="fu">read.delim</span>(<span class="st">"FULL_count_matrix.txt"</span>, <span class="at">header=</span>T, <span class="at">row.names=</span><span class="dv">1</span>) </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># View the count matrix</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">#View(counttable_original)</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene symbol as the identifier (when compared to ENSG ID)</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>counttable<span class="ot">&lt;-</span>counttable_original[,<span class="fu">c</span>(<span class="st">"Symbol"</span>,<span class="st">"WT1"</span>,<span class="st">"WT2"</span>,<span class="st">"WT3"</span>,<span class="st">"KO1"</span>,<span class="st">"KO2"</span>,<span class="st">"KO3"</span>)]</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(counttable) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert Column'GeneSymbol' to rowname)</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(counttable) <span class="ot">&lt;-</span> counttable<span class="sc">$</span>Symbol</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>counttable<span class="ot">&lt;-</span>counttable[,<span class="fu">c</span>(<span class="st">"WT1"</span>,<span class="st">"WT2"</span>,<span class="st">"WT3"</span>,<span class="st">"KO1"</span>,<span class="st">"KO2"</span>,<span class="st">"KO3"</span>)]</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co">#View(counttable)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">The count matrix</figcaption><p></p>
<p><img src="../fig/count_matrix.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="key-points" class="level3 keypoints">
<h3 class="anchored" data-anchor-id="key-points"><strong>Key points</strong></h3>
<ul>
<li>RStudio server can be run on the Nimbus VM instance. This allowes the user an end-to-end experience when analysing the RNA-seq data on the Pawsey resource.</li>
</ul>
</section>


</section>
</section>

<p>All materials copyright Sydney Informatics Hub, University of Sydney</p></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>