<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>RNA-Seq using Nextflow â€“ loadcountmatrix_and_explore</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../lesson.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">RNA-Seq using Nextflow</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html">Home</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../setup.html">Setup</a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-nfcore-rnaseq-to-rstudio" role="button" data-bs-toggle="dropdown" aria-expanded="false">nfcore-rnaseq to RStudio</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-nfcore-rnaseq-to-rstudio">    
        <li>
    <a class="dropdown-item" href="../notebooks/1-recap_Day1.html">
 <span class="dropdown-text">RECAP Day 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/1.2_Learning_objectives.html">
 <span class="dropdown-text">Learning objectives</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/2_nextflow_to_RStudio.html">
 <span class="dropdown-text">Start RStudio on Nimbus instance</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3-LoadCountMatrix_and_explore.html">
 <span class="dropdown-text">The count matrix in R</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-differential-expression-(de-analysis" role="button" data-bs-toggle="dropdown" aria-expanded="false">Differential expression (DE) analysis</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-differential-expression-(de-analysis">    
        <li>
    <a class="dropdown-item" href="../notebooks/4-DE_analysis.html">
 <span class="dropdown-text">Create the DESeqDataSet object in DeSeq2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/4-DE_analysis_2.html">
 <span class="dropdown-text">Identify differentially expressed genes</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-enrichment-analysis" role="button" data-bs-toggle="dropdown" aria-expanded="false">Enrichment analysis</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-enrichment-analysis">    
        <li>
    <a class="dropdown-item" href="../notebooks/6_Whatis_Enrichment_analysis.html">
 <span class="dropdown-text">What is enrichment analysis?</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/7_R_DEGenes_to_functional_enrichments_clusProf.html">
 <span class="dropdown-text">DE genes to functional enrichments</span></a>
  </li>  
    </ul>
  </li>
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#load-count-matrix-and-explore" id="toc-load-count-matrix-and-explore" class="nav-link active" data-scroll-target="#load-count-matrix-and-explore">Load count matrix and explore</a>
  <ul class="collapse">
  <li><a href="#objectives" id="toc-objectives" class="nav-link" data-scroll-target="#objectives"><strong>Objectives</strong></a></li>
  <li><a href="#questions" id="toc-questions" class="nav-link" data-scroll-target="#questions"><strong>Questions</strong></a></li>
  <li><a href="#the-gene-count-matrix" id="toc-the-gene-count-matrix" class="nav-link" data-scroll-target="#the-gene-count-matrix">The gene-count matrix</a></li>
  <li><a href="#working-in-rstudio" id="toc-working-in-rstudio" class="nav-link" data-scroll-target="#working-in-rstudio">Working in RStudio</a></li>
  <li><a href="#r-markdown" id="toc-r-markdown" class="nav-link" data-scroll-target="#r-markdown">R-markdown</a></li>
  <li><a href="#open-a-r-markdown-file" id="toc-open-a-r-markdown-file" class="nav-link" data-scroll-target="#open-a-r-markdown-file">Open a R-markdown file</a></li>
  <li><a href="#exploratory-analysis-and-visualization" id="toc-exploratory-analysis-and-visualization" class="nav-link" data-scroll-target="#exploratory-analysis-and-visualization">Exploratory analysis and visualization</a></li>
  <li><a href="#challenge-1" id="toc-challenge-1" class="nav-link" data-scroll-target="#challenge-1">Challenge #1</a></li>
  <li><a href="#challenge-2" id="toc-challenge-2" class="nav-link" data-scroll-target="#challenge-2">Challenge #2</a></li>
  <li><a href="#principal-component-analysis-pca" id="toc-principal-component-analysis-pca" class="nav-link" data-scroll-target="#principal-component-analysis-pca">Principal Component Analysis (PCA)</a></li>
  <li><a href="#challenge-3" id="toc-challenge-3" class="nav-link" data-scroll-target="#challenge-3">Challenge #3</a></li>
  <li><a href="#why-choose-best-the-dimension" id="toc-why-choose-best-the-dimension" class="nav-link" data-scroll-target="#why-choose-best-the-dimension">Why choose best the dimension?</a></li>
  <li><a href="#a-scree-plot-can-assist" id="toc-a-scree-plot-can-assist" class="nav-link" data-scroll-target="#a-scree-plot-can-assist">A Scree plot can assist</a></li>
  <li><a href="#key-points" id="toc-key-points" class="nav-link" data-scroll-target="#key-points"><strong>Key points</strong></a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">



<section id="load-count-matrix-and-explore" class="level1">
<h1>Load count matrix and explore</h1>
<section id="objectives" class="level3 objectives">
<h3 class="anchored" data-anchor-id="objectives"><strong>Objectives</strong></h3>
<ul>
<li>Load the count matrix in R and perform initial exploratory analysis</li>
</ul>
</section>
<section id="questions" class="level3 questions">
<h3 class="anchored" data-anchor-id="questions"><strong>Questions</strong></h3>
<ul>
<li>How to import a count-matrix in R?</li>
<li>How to perform exploratory analysis of the count-matrix?</li>
</ul>
</section>
<section id="the-gene-count-matrix" class="level3">
<h3 class="anchored" data-anchor-id="the-gene-count-matrix">The gene-count matrix</h3>
<ul>
<li>The gene-count matrix was generated in the below folder using the nfcore-rnaseq pipeline:</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>~/base_directory/working_directory/results/star_salmon/salmon.merged.gene_counts.tsv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li><p>The input to this pipeline were sub-setted fastq files from 6 samples which were mapped to the <code>chr18</code> reference of mouse. This was done so that the excecution of the <code>nfcore-rnaseq</code> pipeline can be demonstrated on our small-sized training VMs <code>(2 cpus and 8 GB RAM)</code> in quick time <code>(~ 20 mins)</code>.</p></li>
<li><p>However, for todayâ€™s session, we will be using a pre-computed gene-count matrix which was generated by aligning the full set of reads to the complete mouse genome.</p></li>
<li><p>This file was downloaded previously along with the other files from cloudstor and is named as <code>Full_count_matrix.txt</code>. It is available in the path as</p></li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>~/base_path/working_directory/Full_count_matrix.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="working-in-rstudio" class="level3">
<h3 class="anchored" data-anchor-id="working-in-rstudio">Working in RStudio</h3>
</section>
<section id="r-markdown" class="level3">
<h3 class="anchored" data-anchor-id="r-markdown">R-markdown</h3>
<ul>
<li>R Markdown is a file format for making dynamic documents with R.</li>
<li>An R Markdown document is written in markdown (an easy-to-write plain text format) and contains chunks of embedded R code, like the document shown below.</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">RStudio interface</figcaption><p></p>
<p><img src="../fig/R_markdown.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="open-a-r-markdown-file" class="level3">
<h3 class="anchored" data-anchor-id="open-a-r-markdown-file">Open a R-markdown file</h3>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">Open Rmarkdown file</figcaption><p></p>
<p><img src="../fig/Rmarkdown_open_file.png" class="img-fluid figure-img"></p>
</figure>
</div>
<ul>
<li>From here on we will be running individual chunks of code in the R-markdown document and follow</li>
</ul>
<section id="load-the-r-libraries" class="level4">
<h4 class="anchored" data-anchor-id="load-the-r-libraries">Load the R-libraries</h4>
<ul>
<li>Import all the R libraries/packages which will assist us at various steps in the analysis.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>({</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"DESeq2"</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"edgeR"</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"limma"</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"RColorBrewer"</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"gplots"</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"EnhancedVolcano"</span>)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"factoextra"</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"devtools"</span>)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"rstudioapi"</span>)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tibble"</span>)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"pheatmap"</span>)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"biomaRt"</span>)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"annotables"</span>)</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"org.Mm.eg.db"</span>)</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"biobroom"</span>)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"clusterProfiler"</span>)</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"pathfindR"</span>)</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="set-the-current-working-directory" class="level4">
<h4 class="anchored" data-anchor-id="set-the-current-working-directory">Set the current working directory</h4>
<div class="sourceCode" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>current_path <span class="ot">&lt;-</span> <span class="fu">getActiveDocumentContext</span>()<span class="sc">$</span>path </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="fu">dirname</span>(current_path ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="import-the-count-matrix" class="level4">
<h4 class="anchored" data-anchor-id="import-the-count-matrix">Import the count matrix</h4>
<ul>
<li>The DESeq2 package expects count data from RNA-seq or another high-throughput sequencing experiment in the form of a matrix of integer values, as input.</li>
<li>The value in the i-th row and the j-th column of the matrix tells how many reads can be assigned to gene i in sample j.</li>
<li>The values in the matrix should be un-normalized counts or estimated counts of sequencing reads (for single-end RNA-seq) or fragments (for paired-end RNA-seq).</li>
<li>The DESeq2 model internally corrects for library size, so transformed or normalized values such as counts scaled by library size should NOT be used as input.</li>
<li>Next we read in the count matrix generated as inout the R-script.</li>
<li>A gene-count matrix was generated in the folder</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>~/base_directory/working_directory/results/star_salmon/salmon.merged.gene_counts.tsv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>using the nfcore-rnaseq pipeline. The input to this pipeline were sub-setted fastq files from 6 samples which were mapped to the chr18 reference of mouse. This was done so that the excecution of the nfcore-rnaseq pipeline can be demonstrated on our small-sized training VMs <code>(2 cpus and 8 GB RAM)</code> in quick time <code>(~ 20 mins)</code>. - However, for todayâ€™s session, we will be using a pre-computed gene-count matrix which was generated by aligning the full set of reads to the complete mouse genome. - This file was downloaded previously along with the other files from cloudstor and is named as <code>Full_count_matrix.txt</code>. It is available in the path as <code>~/base_path/working_directory/Full_count_matrix.txt</code>. - Run the command</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>cp ~/base_directory/working_directory/FULL_count_matrix.txt ~/base_directory/working_directory/rstudio/</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the Full count matrix (File has been provided for download)</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>counttable_original<span class="ot">&lt;-</span><span class="fu">read.delim</span>(<span class="st">"FULL_count_matrix.txt"</span>, <span class="at">header=</span>T, <span class="at">row.names=</span><span class="dv">1</span>) </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># View the count matrix</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">#View(counttable_original)</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene symbol as the identifier (when compared to ENSG ID)</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>counttable<span class="ot">&lt;-</span>counttable_original[,<span class="fu">c</span>(<span class="st">"Symbol"</span>,<span class="st">"WT1"</span>,<span class="st">"WT2"</span>,<span class="st">"WT3"</span>,<span class="st">"KO1"</span>,<span class="st">"KO2"</span>,<span class="st">"KO3"</span>)]</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(counttable) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert Column'GeneSymbol' to rowname)</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(counttable) <span class="ot">&lt;-</span> counttable<span class="sc">$</span>Symbol</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>counttable<span class="ot">&lt;-</span>counttable[,<span class="fu">c</span>(<span class="st">"WT1"</span>,<span class="st">"WT2"</span>,<span class="st">"WT3"</span>,<span class="st">"KO1"</span>,<span class="st">"KO2"</span>,<span class="st">"KO3"</span>)]</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co">#View(counttable)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">count matrix</figcaption><p></p>
<p><img src="../fig/count_matrix.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="exploratory-analysis-and-visualization" class="level3">
<h3 class="anchored" data-anchor-id="exploratory-analysis-and-visualization">Exploratory analysis and visualization</h3>
<section id="box-plot" class="level4">
<h4 class="anchored" data-anchor-id="box-plot">Box plot</h4>
<ul>
<li>A quick look at distribution of the data across sample using box plots.</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">log2</span>((counttable)<span class="sc">+</span><span class="dv">1</span>),<span class="at">las=</span><span class="dv">3</span>, <span class="at">col=</span><span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">Box plot : before applying transformation</figcaption><p></p>
<p><img src="../fig/box_plot_orig.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="what-is-a-deseq2-object" class="level4">
<h4 class="anchored" data-anchor-id="what-is-a-deseq2-object">What is a DESeq2 object?</h4>
<ul>
<li>The basic entity used by the DESeq2 package is and object class called DESeqDataSet.</li>
<li>The DESeqDataSet object is used to store both the read counts and the intermediate estimated quantities during statistical analysis</li>
<li>We will call this object by name â€˜ddsâ€™ which is a standard practice (<strong>Note</strong> We can name this object absolutely anything in real world ðŸ™‚).</li>
<li>Please see the <a href="http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">guide</a> for detailed information about the DESeq2 library.</li>
<li>We also define a condition variable to associate the individual columns (samples) in the matrix to their appropriate experimental condition (either â€˜Wild-typeâ€™ or â€˜Knockoutâ€™).</li>
<li>We will then prepare the DESeq2 object with <code>design = ~1</code>;
<ul>
<li>A design of ~1 is used for no experimental design and is useful for exploring quality of the data (not for identifying differentially expressed -DE genes).</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a condition variable </span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>condition<span class="ot">=</span><span class="fu">c</span>(<span class="st">"Wild"</span>,<span class="st">"Wild"</span>,<span class="st">"Wild"</span>,<span class="st">"KO"</span>,<span class="st">"KO"</span>,<span class="st">"KO"</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row.names=</span><span class="fu">colnames</span>(counttable),condition)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#View(meta)</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counttable, </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="at">colData =</span> meta, </span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="at">design =</span> <span class="sc">~</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="data-transformations" class="level4">
<h4 class="anchored" data-anchor-id="data-transformations">Data transformations</h4>
<ul>
<li><p>For visualization or clustering it is useful to work with transformed versions of the count data.</p></li>
<li><p>The two most common choices of transformation used for this purpose are -</p>
<ul>
<li>Variance stabilizing transformation (vst)</li>
<li>Regularized log transformation (rlog)</li>
</ul></li>
<li><p>These algorithms transform the raw count data (which is heteroskedatic - variance grows with the mean) into homoskedatic data (variance is not dependant on the mean).</p></li>
<li><p>Both methods produce data on the log2 scale, and normalize for other factors such as library size. Setting <code>blind=TRUE</code> (the default) should be used to compare samples in a manner wholly unbiased about the information about experimental groups, for example to perform sample QC.</p></li>
<li><p><strong>Note</strong> : In order to test for differential expression, we operate on raw counts (and not normalized/transformed counts).</p></li>
</ul>
</section>
<section id="variance-stabilisting-transformation-vst" class="level4">
<h4 class="anchored" data-anchor-id="variance-stabilisting-transformation-vst">Variance stabilisting transformation (VST)</h4>
<ul>
<li>VST by default uses a subset of 1000 rows to estimate the dispersion trend.</li>
<li>This method is much faster than the other transformation mathod called - rlog.</li>
<li>Hence, VST is recommended if you have hundreds of samples.</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>vst <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>vst.data <span class="ot">&lt;-</span> <span class="fu">assay</span>(vst)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Regularized log (rlog) takes a long time with 50 or more samples</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="co"># rld &lt;- rlog(dds, blind=FALSE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="re-drawing-the-box-plots" class="level4">
<h4 class="anchored" data-anchor-id="re-drawing-the-box-plots">Re-drawing the box plots</h4>
<ul>
<li>We will now re-generate the box plots to see if the transformation has brought about any difference in distribution.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(vst.data,<span class="at">las=</span><span class="dv">3</span>, <span class="at">col=</span><span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">Box plot : after applying vst transformation</figcaption><p></p>
<p><img src="../fig/box_plot_transformed.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="challenge-1" class="level3 challenge">
<h3 class="anchored" data-anchor-id="challenge-1">Challenge #1</h3>
<p>Looking at the above two box plots:</p>
<ul>
<li>Do you see a difference in the two box plots (before and after transformation)?</li>
<li>Do you think the transformation has helped?</li>
</ul>
<details>
<summary>
Solution
</summary>
<p>This is my solution</p>
</details>
</section>
<section id="heatmap-sample-to-sample-distances" class="level4">
<h4 class="anchored" data-anchor-id="heatmap-sample-to-sample-distances">Heatmap (Sample-to-sample distances)</h4>
<ul>
<li>A heatmap of a distance matrix gives us an overview over similarities and dissimilarities between samples.</li>
<li>We have to provide a hierarchical clustering hc to the heatmap function based on the sample distances.</li>
</ul>
<div class="sourceCode" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>sampleDists <span class="ot">&lt;-</span> <span class="fu">dist</span>(<span class="fu">t</span>(<span class="fu">assay</span>(vst)))</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>sampleDistMatrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(sampleDists)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sampleDistMatrix) <span class="ot">&lt;-</span> <span class="fu">paste</span>(vst<span class="sc">$</span>condition, vst<span class="sc">$</span>type, <span class="at">sep=</span><span class="st">"-"</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sampleDistMatrix) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>( <span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Blues"</span>)) )(<span class="dv">255</span>)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(sampleDistMatrix,</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows=</span>sampleDists,</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_cols=</span>sampleDists,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">col=</span>colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">Heatmap to view distances between samples</figcaption><p></p>
<p><img src="../fig/Heatmap_sample_distances.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="challenge-2" class="level3 challenge">
<h3 class="anchored" data-anchor-id="challenge-2">Challenge #2</h3>
<p>Does the heatmap convey information about the samples and the two conditions?</p>
<details>
<summary>
Solution
</summary>
<ul>
<li>The WT samples cluster together.</li>
<li>There is a possibility that one of the knockout samples is a bit distant to itâ€™s replicates. Need to investigate further.</li>
</ul>
</details>
</section>
<section id="principal-component-analysis-pca" class="level3">
<h3 class="anchored" data-anchor-id="principal-component-analysis-pca">Principal Component Analysis (PCA)</h3>
<ul>
<li>PCA is a linear dimensionality reduction technique. It extracts information from a high-dimensional space by projecting it into a lower-dimensional sub-space. Here dimensions are the features (all your thousands of genes from the genome) which represent the data and explain most of the variability.</li>
<li>PCA preserves the essential parts that have more variation of the data and remove the non-essential parts with fewer variation.</li>
<li>This plot is useful for visualizing the overall effect of experimental covariates and batch effects.</li>
<li>Ideally, in your experiment, the most of the variability is explained by the conditions (e.g.&nbsp;WT versus KO in this case). If any of the samples group differently, the DE analysis cannot be relied upon.</li>
<li>If we have big number of replicate samples, PCA allows you to identify and remove (if required) any outliers.</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vst)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">picture alt</figcaption><p></p>
<p><img src="../fig/PCA_noLegend.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="challenge-3" class="level3 challenge">
<h3 class="anchored" data-anchor-id="challenge-3">Challenge #3</h3>
<ul>
<li>Looking at just the PCA, can you guess which samples belong to which conditions?</li>
<li>Why?</li>
</ul>
<details>
<summary>
Solution
</summary>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">picture alt</figcaption><p></p>
<p><img src="../fig/PCA.png" class="img-fluid figure-img"></p>
</figure>
</div>
</details>
</section>
<section id="why-choose-best-the-dimension" class="level3">
<h3 class="anchored" data-anchor-id="why-choose-best-the-dimension">Why choose best the dimension?</h3>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">picture alt</figcaption><p></p>
<p><img src="../fig/Prince_William_dimensionality.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vst)<span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="a-scree-plot-can-assist" class="level3">
<h3 class="anchored" data-anchor-id="a-scree-plot-can-assist">A Scree plot can assist</h3>
<ul>
<li>The Principal Component matrix has the same dimensions as the original data matrix; however, many of the PCs may not be informative. A reduction in the number of PCs is typically required.</li>
<li>To do so, it is useful to examine the amount of variance explained by each new PC vector.</li>
<li>A scree plot is a line plot of the principal components in an analysis.
<ul>
<li>The scree plot is used to determine the number of principal components to keep in a principal component analysis (PCA).</li>
<li>It shows the eigenvalues on the y-axis and the number of factors on the x-axis and it always displays a downward curve.</li>
<li>The point where the slope of the curve is clearly leveling off (the â€œelbow) indicates the number of factors that should be generated by the analysis.</li>
<li>Letâ€™s draw a scree plot.</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>pca<span class="ot">=</span><span class="fu">prcomp</span>(<span class="fu">t</span>(<span class="fu">assay</span>(vst)),<span class="at">scale=</span><span class="cn">FALSE</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">repr.plot.width=</span><span class="fl">0.5</span>, <span class="at">repr.plot.height=</span><span class="fl">0.5</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_eig</span>(pca, <span class="at">addlabels =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">picture alt</figcaption><p></p>
<p><img src="../fig/scree_plot.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>We can see that PC1 followed by PC2 have captured most of the variance.</p>
<section id="small-contributions-of-many-genes-to-variation" class="level4">
<h4 class="anchored" data-anchor-id="small-contributions-of-many-genes-to-variation">Small contributions of many genes to variation</h4>
<p>It is observed that in biological experiments multiple genes contribute to the the principle compoments in a small way. This can be seen in the plot below where the Y-axis shows the small % contributions of multiple genes to PC1</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># The function fviz_contrib() [in factoextra package] is used </span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="co"># to visualize the contributions of rows/columns from the </span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co"># results of Principal Component Analysis (PCA)</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_contrib</span>(pca, <span class="at">choice =</span> <span class="st">"var"</span>, <span class="at">axes =</span> <span class="dv">1</span>, <span class="at">top =</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">picture alt</figcaption><p></p>
<p><img src="../fig/PCA1_gene_contributions.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="key-points" class="level3 keypoints">
<h3 class="anchored" data-anchor-id="key-points"><strong>Key points</strong></h3>
<ul>
<li>This is a key point</li>
<li>Another one</li>
</ul>
</section>


</section>

<p>All materials copyright Sydney Informatics Hub, University of Sydney</p></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>