<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.251">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>RNA-Seq using Nextflow: Day 2 – r_degenes_to_functional_enrichments_clusprof</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../lesson.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">RNA-Seq using Nextflow: Day 2</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="https://sydney-informatics-hub.github.io/training.RNAseq.series-quarto/">Home</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://sydney-informatics-hub.github.io/training.RNAseq.series-quarto/setup.html">Setup</a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-day-2-kick-off" role="button" data-bs-toggle="dropdown" aria-expanded="false">Day 2 kick off</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-day-2-kick-off">    
        <li>
    <a class="dropdown-item" href="../notebooks/0.0_recap_Day1.html">
 <span class="dropdown-text">Recap Day 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/0.1_recap_casestudy.html">
 <span class="dropdown-text">Revisit the case study</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/0.2_Learning_objectives.html">
 <span class="dropdown-text">Learning objectives</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/0.3_nextflow_to_RStudio.html">
 <span class="dropdown-text">Set up RStudio</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-gene-counts-to-differentially-expressed-genes" role="button" data-bs-toggle="dropdown" aria-expanded="false">Gene counts to differentially expressed genes</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-gene-counts-to-differentially-expressed-genes">    
        <li>
    <a class="dropdown-item" href="../notebooks/1.0_exploratory_analysis.html">
 <span class="dropdown-text">Exploratory analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/1.1_DE_analysis.html">
 <span class="dropdown-text">Differential expression</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/1.2_Whatis_Enrichment_analysis.html">
 <span class="dropdown-text">Enrichment analysis</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../notebooks/2.0_Day2_wrapup.html">Day 2 wrap up</a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#de-genes-to-functional-enrichment-in-r" id="toc-de-genes-to-functional-enrichment-in-r" class="nav-link active" data-scroll-target="#de-genes-to-functional-enrichment-in-r"><strong>DE genes to functional enrichment in R</strong></a>
  <ul class="collapse">
  <li><a href="#questions" id="toc-questions" class="nav-link" data-scroll-target="#questions"><strong>Questions</strong></a></li>
  <li><a href="#objectives" id="toc-objectives" class="nav-link" data-scroll-target="#objectives"><strong>Objectives</strong></a></li>
  <li><a href="#gene-ontology-go-over-representation-analysis" id="toc-gene-ontology-go-over-representation-analysis" class="nav-link" data-scroll-target="#gene-ontology-go-over-representation-analysis"><strong>Gene Ontology (GO) over-representation analysis</strong></a>
  <ul class="collapse">
  <li><a href="#prepare-input-and-filter-for-up-and-down-regulated-genes" id="toc-prepare-input-and-filter-for-up-and-down-regulated-genes" class="nav-link" data-scroll-target="#prepare-input-and-filter-for-up-and-down-regulated-genes"><strong>Prepare input and filter for up and down regulated genes</strong></a></li>
  <li><a href="#genes-down-regulated-in-wt" id="toc-genes-down-regulated-in-wt" class="nav-link" data-scroll-target="#genes-down-regulated-in-wt"><strong>Genes Down-regulated in WT</strong></a></li>
  <li><a href="#challenge-1" id="toc-challenge-1" class="nav-link" data-scroll-target="#challenge-1"><strong>Challenge #1</strong></a></li>
  <li><a href="#challenge-2" id="toc-challenge-2" class="nav-link" data-scroll-target="#challenge-2"><strong>Challenge #2</strong></a></li>
  <li><a href="#challenge-3" id="toc-challenge-3" class="nav-link" data-scroll-target="#challenge-3"><strong>Challenge #3</strong></a></li>
  <li><a href="#key-points" id="toc-key-points" class="nav-link" data-scroll-target="#key-points"><strong>Key points</strong></a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">



<section id="de-genes-to-functional-enrichment-in-r" class="level1">
<h1><strong>DE genes to functional enrichment in R</strong></h1>
<section id="questions" class="level3 questions">
<h3 class="anchored" data-anchor-id="questions"><strong>Questions</strong></h3>
<ul>
<li>How to perform enrichment analysis in R (RStudio)?</li>
<li>How to visualise functionally enriched Gene ontologies (GO) / pathways as networks?</li>
</ul>
</section>
<section id="objectives" class="level3 objectives">
<h3 class="anchored" data-anchor-id="objectives"><strong>Objectives</strong></h3>
<ul>
<li>Perform Functional enrichment analysis of the DE genes in <code>R</code></li>
</ul>
</section>
<section id="gene-ontology-go-over-representation-analysis" class="level2">
<h2 class="anchored" data-anchor-id="gene-ontology-go-over-representation-analysis"><strong>Gene Ontology (GO) over-representation analysis</strong></h2>
<ul>
<li>We will be using the R-package clusterProfiler to perform over-representation analysis on GO terms.</li>
<li>The tool takes as input a list of significant genes (DEGs in this case) and a background gene list to perform statistical enrichment analysis using hypergeometric testing.</li>
<li>The basic arguments allow the user to select the appropriate organism and GO ontology (BP, CC, MF) to test.</li>
</ul>
<section id="prepare-input-and-filter-for-up-and-down-regulated-genes" class="level3">
<h3 class="anchored" data-anchor-id="prepare-input-and-filter-for-up-and-down-regulated-genes"><strong>Prepare input and filter for up and down regulated genes</strong></h3>
<ul>
<li>Filter by padjust and log fold change.</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># P adj &lt; 0.05 </span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">&lt;-</span> res_tidy.DE[res_tidy.DE<span class="sc">$</span>p.adjusted <span class="sc">&lt;</span> <span class="fl">0.05</span>, ]</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Upregulated: LFC &gt; 1, remove NAs</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>sig.up <span class="ot">&lt;-</span> sig[sig<span class="sc">$</span>estimate <span class="sc">&gt;</span> <span class="dv">1</span>, ]</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>sig.up <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(sig.up)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>sig.up.LFC <span class="ot">&lt;-</span> sig.up<span class="sc">$</span>estimate</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sig.up.LFC) <span class="ot">&lt;-</span> sig.up<span class="sc">$</span>gene</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort by LFC, decreasing</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>sig.up.LFC <span class="ot">&lt;-</span> <span class="fu">sort</span>(sig.up.LFC, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Downregulated: LFC &lt; 1, remove NAs</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>sig.dn <span class="ot">&lt;-</span> sig[sig<span class="sc">$</span>estimate <span class="sc">&lt;</span> <span class="dv">1</span>, ]</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>sig.dn <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(sig.dn)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>sig.dn.LFC <span class="ot">&lt;-</span> sig.dn<span class="sc">$</span>estimate</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sig.dn.LFC) <span class="ot">&lt;-</span> sig.dn<span class="sc">$</span>gene</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort by LFC, decreasing</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>sig.dn.LFC <span class="ot">&lt;-</span> <span class="fu">sort</span>(sig.dn.LFC, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="genes-down-regulated-in-wt" class="level3">
<h3 class="anchored" data-anchor-id="genes-down-regulated-in-wt"><strong>Genes Down-regulated in WT</strong></h3>
<section id="the-function-enrichgo" class="level4">
<h4 class="anchored" data-anchor-id="the-function-enrichgo"><strong>The function enrichGO()</strong></h4>
<p>The clusterProfiler package implements enrichGO() for gene ontology over-representation test.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>ego.up <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> <span class="fu">names</span>(sig.up.LFC),</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">OrgDb =</span> org.Mm.eg.db, </span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">keyType =</span> <span class="st">'SYMBOL'</span>,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">readable =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ont =</span> <span class="st">"ALL"</span>,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>, </span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>                      <span class="at">qvalueCutoff =</span> <span class="fl">0.2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="bar-plot" class="level4">
<h4 class="anchored" data-anchor-id="bar-plot"><strong>Bar plot</strong></h4>
<ul>
<li>Bar plot is the most widely used method to visualize enriched terms.</li>
<li>It depicts the enrichment scores (e.g.&nbsp;p-values) and gene count or ratio as bar height and color.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(ego.up, <span class="at">showCategory=</span><span class="dv">20</span>){width<span class="ot">=</span><span class="dv">70</span>%}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">Bar Plot: Genes DOWN in WT</figcaption><p></p>
<p><img src="../fig/BarPlot_DownInWT.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="dot-plot" class="level4">
<h4 class="anchored" data-anchor-id="dot-plot"><strong>Dot plot</strong></h4>
<ul>
<li>A Dot plot is similar to a scatter plot and bar plot with the capability to encode another score as dot size.</li>
<li>In R the dot plot displays the index (each category) in the vertical axis and the corresponding value in the horizontal axis, so you can see the value of each observation following a horizontal line from the label.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(ego.up, <span class="at">showCategory=</span><span class="dv">20</span>,<span class="at">font.size =</span> <span class="dv">10</span>){width<span class="ot">=</span><span class="dv">70</span>%}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">Dot Plot: Genes DOWN in WT</figcaption><p></p>
<p><img src="../fig/DotPlot_DownInWT.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="cnetplot" class="level4">
<h4 class="anchored" data-anchor-id="cnetplot"><strong>cnetplot</strong></h4>
<ul>
<li>Both the barplot and dotplot only displayed most significant enriched terms, while users may want to know which genes are involved in these significant terms.</li>
<li>The cnetplot depicts the linkages of genes and biological concepts (e.g.&nbsp;GO terms or KEGG pathways) as a network.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cnetplot</span>(ego.up, </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a> <span class="at">categorySize=</span><span class="st">"pvalue"</span>, </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a> <span class="at">foldChange=</span>sig.up.LFC,</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a> <span class="at">cex_label_gene =</span> <span class="dv">1</span>,</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a> <span class="at">showCategory =</span> <span class="dv">5</span>,<span class="at">cex_label_category=</span><span class="fl">1.2</span>,<span class="at">shadowtext=</span><span class="st">'category'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">cnetPlot: Genes DOWN in WT</figcaption><p></p>
<p><img src="../fig/cnetPlot_DownInWT.png" class="img-fluid figure-img" style="width:70.0%"></p>
</figure>
</div>
</section>
<section id="heatmap-like-functional-classification" class="level4">
<h4 class="anchored" data-anchor-id="heatmap-like-functional-classification"><strong>Heatmap-like functional classification</strong></h4>
<ul>
<li>The heatplot is similar to cnetplot, while displaying the relationships as a heatmap.</li>
<li>The gene-concept network may become too complicated if user want to show a large number significant terms.</li>
<li>The heatplot can simplify the result and more easy to identify expression patterns.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">heatplot</span>(ego.up)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">Heatplot: Genes DOWN in WT</figcaption><p></p>
<p><img src="../fig/heatMap_DownInWT.png" class="img-fluid figure-img" style="width:70.0%"></p>
</figure>
</div>
</section>
</section>
<section id="challenge-1" class="level3 challenge">
<h3 class="anchored" data-anchor-id="challenge-1"><strong>Challenge #1</strong></h3>
<ul>
<li>Do you find any of the top enriched GO categories relavant to the experiment?</li>
<li>If so what is your deduction? Do you feel RNA-seq experiment has helped ?</li>
</ul>
<details>
<summary>
Solution
</summary>
<p>My Solution here</p>
</details>
</section>
<section id="challenge-2" class="level3 challenge">
<h3 class="anchored" data-anchor-id="challenge-2"><strong>Challenge #2</strong></h3>
<ul>
<li>Can you perform similar analysis for genes which are UP-regulated in WT?</li>
</ul>
<details>
<summary>
Solution
</summary>
<section id="genes-up-regulated-in-wt" class="level5">
<h5 class="anchored" data-anchor-id="genes-up-regulated-in-wt"><strong>Genes Up-regulated in WT</strong></h5>
<div class="sourceCode" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>ego.dn <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> <span class="fu">names</span>(sig.dn.LFC),</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="at">OrgDb =</span> org.Mm.eg.db, </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="at">keyType =</span> <span class="st">'SYMBOL'</span>,</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="at">readable =</span> <span class="cn">FALSE</span>,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="at">ont =</span> <span class="st">"ALL"</span>,</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>, </span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="at">qvalueCutoff =</span> <span class="fl">0.2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="bar-plot-1" class="level4">
<h4 class="anchored" data-anchor-id="bar-plot-1"><strong>Bar plot</strong></h4>
<div class="sourceCode" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(ego.dn, <span class="at">showCategory=</span><span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">Bar Plot: Genes UP in WT</figcaption><p></p>
<p><img src="../fig/BarPlot_UpInWT.png" class="img-fluid figure-img" style="width:70.0%"></p>
</figure>
</div>
</section>
<section id="dot-plot-1" class="level4">
<h4 class="anchored" data-anchor-id="dot-plot-1"><strong>Dot-plot</strong></h4>
<div class="sourceCode" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(ego.dn, <span class="at">showCategory=</span><span class="dv">20</span>,<span class="at">font.size =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">Dot Plot: Genes UP in WT</figcaption><p></p>
<p><img src="../fig/DotPlot_UpInWT.png" class="img-fluid figure-img" style="width:70.0%"></p>
</figure>
</div>
</section>
<section id="cnetplot-1" class="level4">
<h4 class="anchored" data-anchor-id="cnetplot-1"><strong>cnetplot</strong></h4>
<div class="sourceCode" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cnetplot</span>(ego.dn, </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a> <span class="at">categorySize=</span><span class="st">"pvalue"</span>, </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a> <span class="at">foldChange=</span>sig.dn.LFC,</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a> <span class="at">cex_label_gene =</span> <span class="fl">0.7</span>,</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a> <span class="at">showCategory =</span> <span class="dv">5</span>,<span class="at">cex_label_category=</span><span class="fl">1.5</span>,<span class="at">shadowtext=</span><span class="st">'category'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">cnetPlot: Genes UP in WT</figcaption><p></p>
<p><img src="../fig/cnetPlot_UpInWT.png" class="img-fluid figure-img" style="width:70.0%"></p>
</figure>
</div>
</section>
<section id="heatmap-like-functional-classification-1" class="level4">
<h4 class="anchored" data-anchor-id="heatmap-like-functional-classification-1"><strong>Heatmap-like functional classification</strong></h4>
<div class="sourceCode" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">heatplot</span>(ego.dn)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption">Heatplot: Genes UP in WT</figcaption><p></p>
<p><img src="../fig/heatMap_UpInWT.png" class="img-fluid figure-img" style="width:70.0%"></p>
</figure>
</div>
</section></details>
</section>
</section>
<section id="challenge-3" class="level3 challenge">
<h3 class="anchored" data-anchor-id="challenge-3"><strong>Challenge #3</strong></h3>
<ul>
<li>What about the GO categories for the genes found UP-regulated in WT? Any interesting find?</li>
</ul>
<details>
<summary>
Solution
</summary>
<p>My Solution here</p>
</details>
</section>
<section id="key-points" class="level3 keypoints">
<h3 class="anchored" data-anchor-id="key-points"><strong>Key points</strong></h3>
<ul>
<li>This is a key point</li>
<li>Another one</li>
</ul>
</section>


</section>


<p>All materials copyright Sydney Informatics Hub, University of Sydney</p></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>